Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp

```diff
@@ -70,10 +70,7 @@ Ref<Nicosia::Buffer> CoordinatedGraphicsLayer::paintTile(const IntRect& tileRect
 
     // Skia/GPU - accelerated rendering.
     if (auto* acceleratedBufferPool = m_coordinator->skiaAcceleratedBufferPool()) {
-        auto* glContext = PlatformDisplay::sharedDisplayForCompositing().skiaGLContext();
-        RELEASE_ASSERT(glContext);
-        GLContext::ScopedGLContextCurrent scopedCurrent(*glContext);
-
+        PlatformDisplay::sharedDisplayForCompositing().skiaGLContext()->makeContextCurrent();
```

Comment:
No null-checking here? Maybe we can have a safeSkiaGLContext() accessor that returns a reference to manifest that this cannot be non-null, if that's the case. Didn't check on my own, just pondering.

Reply:
Good point. I think we should have the check in CompositingCoordinator where we decide whether to use CPU or GPU, and not create the buffer pool if we failed to create the Skia GL context. I can do it in a follow up.